apps/api/src/app/app.module.ts

```diff
@@ -1,24 +1,39 @@
 import { Logger, Module } from '@nestjs/common'
 import { ConfigModule } from '@nestjs/config'
-import { APP_PIPE } from '@nestjs/core'
-import { ZodValidationPipe } from 'nestjs-zod'
+import { APP_FILTER, APP_INTERCEPTOR, APP_PIPE } from '@nestjs/core'
+import { ZodSerializerInterceptor, ZodValidationPipe } from 'nestjs-zod'
 
 import { DatabaseModule } from '../database/database.module'
 import { MoviesModule } from '../movies/movies.module'
+import { SubscriptionsModule } from '../subscriptions/subscriptions.module'
 
 import { AppController } from './app.controller'
 import { AppService } from './app.service'
 import { validate } from './config/validate'
+import { HttpExceptionFilter } from './http-exception.filter'
 
 @Module({
-  imports: [ConfigModule.forRoot({ isGlobal: true, validate }), DatabaseModule, MoviesModule],
+  imports: [
+    ConfigModule.forRoot({ isGlobal: true, validate }),
+    DatabaseModule,
+    MoviesModule,
+    SubscriptionsModule
+  ],
   controllers: [AppController],
   providers: [
     Logger,
     AppService,
     {
       provide: APP_PIPE,
       useClass: ZodValidationPipe
+    },
+    {
+      provide: APP_INTERCEPTOR,
```

Comment:
So what we're doing here is basically addding a validation middleware of the incoming payload? Or?

Reply:
incoming payload validation would be the block above (`APP_PIPE` and `ZodValidationPipe`), this one is about how you want response payload to be shown